charts/backstage/values.yaml

```diff
@@ -81,26 +81,38 @@ upstream:
         cpu: 1000m
         memory: 2.5Gi
         ephemeral-storage: 5Gi
+    startupProbe:
+      # This gives enough time upon container startup before the liveness and readiness probes are triggered.
+      # Giving (120s = initialDelaySeconds + failureThreshold * periodSeconds) to account for the worst case scenario.
+      httpGet:
+        path: /.backstage/health/v1/liveness
+        port: backend
+        scheme: HTTP
+      initialDelaySeconds: 30
+      timeoutSeconds: 4
```

Comment:
is 4s enough for slow/flaky clusters? why not 10s?

Reply:
High timeouts could hide potential problems and significantly increase error detection time.

I would say that 4 seconds is high enough; if you don't get a response on the health check endpoint within 4 seconds, then something is already seriously wrong.

The default configuration should not be optimized for unstable or flaky clusters. If someone has a cluster like this, they can always customize those settings.

Reply:
Yeah, tbh, a timeout of 4 seconds seemed a lot to me as well. We tried with 2 seconds, but in Patrick's test, we faced an issue with the readiness probe which failed to be reachable on the first check. That's why we thought 4 seconds should be good.